# Introduction to Jupyter Notebook
## Heading Level 2
### Heading Level 3
#### Heading Level 4

In [1]:
#import libraries
import numpy as np
import pandas as pd

In [4]:
#check versions of packages
print("numpy version:", np.__version__)
print("pandas version:", pd.__version__)

numpy version: 1.23.5
pandas version: 1.5.3


## Import Data

In [10]:
#import data as a pandas data frame
df_golfers = pd.read_csv('/Users/kevin/Documents/Bellevue Data Science/DSC310-T301 Data Analysis and Visualization/PGATOURStrokesGained2010-18.csv')
#display first three rows of data
df_golfers.head(3)

Player_Name  Rounds  Fairway_Percentage  Year  Average_Distance    GIR  \
0  Henrik Stenson    60.0               75.19  2018             291.5  73.51   
1     Ryan Armour   109.0               73.58  2018             283.5  68.22   
2     Chez Reavie    93.0               72.24  2018             286.5  68.67   

   Average_Putts  Average_Scrambling  Average_Score  Wins  Top_10 Points  \
0          29.93               60.67         69.617     0       5    868   
1          29.31               60.13         70.758     1       3  1,006   
2          29.12               62.27         70.432     0       3  1,020   

       Money  SG_OTT  SG_APR  SG_ARG  Average_SG_Putts  Average_SG_Total  
0  2680487.0   0.427   0.960  -0.027            -0.207             1.153  
1  2485203.0  -0.012   0.213   0.194            -0.058             0.337  
2  2700018.0   0.183   0.437  -0.137             0.192             0.674

## Explore the Data

In [11]:
#check for missing data
df_golfers.isna().sum()

Player_Name             0
Rounds                634
Fairway_Percentage    634
Year                    0
Average_Distance      634
GIR                   634
Average_Putts         634
Average_Scrambling    634
Average_Score         634
Wins                    0
Top_10                  0
Points                  0
Money                   0
SG_OTT                634
SG_APR                634
SG_ARG                634
Average_SG_Putts      634
Average_SG_Total      634
dtype: int64

In [13]:
#get top 10 golfers by Average_SG_Total
df_golfers.sort_values(by = 'Average_SG_Total', axis = 0, ascending = False).head(10)

Player_Name  Rounds  Fairway_Percentage  Year  Average_Distance  \
1264    Rory McIlroy    60.0               56.61  2012             310.1   
124   Dustin Johnson    77.0               59.46  2018             314.0   
1163     Tiger Woods    69.0               63.93  2012             297.4   
1356     Luke Donald    73.0               64.29  2011             284.1   
861     Rory McIlroy    66.0               59.93  2014             310.5   
579   Henrik Stenson    63.0               69.91  2015             296.4   
931   Steve Stricker    51.0               70.65  2013             283.6   
647    Jordan Spieth    91.0               62.91  2015             291.8   
563        Jason Day    76.0               50.46  2016             304.2   
729        Jason Day    75.0               55.94  2015             313.7   

        GIR  Average_Putts  Average_Scrambling  Average_Score  Wins  Top_10  \
1264  66.36          28.72               60.24         68.873     2       7   
124   70.57          28.47               62.50         68.698     3      10   
1163  67.58          28.91               63.17         68.904     3       6   
1356  67.33          28.03               63.71         68.861     1      10   
861   69.44          28.59               58.52         68.827     3       9   
579   73.52          29.87               62.24         69.354     0       4   
931   71.16          28.68               65.57         68.893     0       5   
647   67.87          27.82               65.03         68.938     4      14   
563   67.23          28.06               61.92         69.313     3       9   
729   70.83          28.44               65.34         69.161     3       8   

     Points       Money  SG_OTT  SG_APR  SG_ARG  Average_SG_Putts  \
1264  2,092   8047952.0   1.072   1.002   0.273             0.058   
124   2,717   8457352.0   0.919   0.829   0.238             0.385   
1163  2,269   6133158.0   0.553   1.224   0.194             0.339   
1356  1,856   6683214.0   0.040   1.094   0.273             0.870   
861   2,582   8280096.0   1.367   0.602   0.022             0.274   
579     952   4755070.0   0.448   1.244   0.082             0.436   
931   1,117   4440532.0   0.311   0.567   0.596             0.720   
647   4,169  12030465.0   0.494   0.618   0.471             0.571   
563   2,735   8045112.0   0.186   0.426   0.375             1.130   
729   2,459   9403330.0   0.772   0.461   0.287             0.586   

      Average_SG_Total  
1264             2.406  
124              2.372  
1163             2.310  
1356             2.278  
861              2.266  
579              2.210  
931              2.193  
647              2.154  
563              2.118  
729              2.106

In [15]:
#get top 10 golfers by Average_SG_Total and displaying only Player_Name, Year, and Average_SG_Total
df_golfers[['Player_Name', 'Year', 'Average_SG_Total']].sort_values(by = 'Average_SG_Total', axis = 0, ascending = False).head(10)

Player_Name  Year  Average_SG_Total
1264    Rory McIlroy  2012             2.406
124   Dustin Johnson  2018             2.372
1163     Tiger Woods  2012             2.310
1356     Luke Donald  2011             2.278
861     Rory McIlroy  2014             2.266
579   Henrik Stenson  2015             2.210
931   Steve Stricker  2013             2.193
647    Jordan Spieth  2015             2.154
563        Jason Day  2016             2.118
729        Jason Day  2015             2.106

In [16]:
#find top 5 golfers by mean Average_SG_Total
df_golfers[['Player_Name', 'Average_SG_Total']].groupby('Player_Name').mean() \
.sort_values(by = 'Average_SG_Total', axis = 0, ascending = False).head(5)

Average_SG_Total
Player_Name                      
Tiger Woods              1.990000
Tommy Fleetwood          1.606000
Steve Stricker           1.551400
Rory McIlroy             1.525143
Jordan Spieth            1.499000

### Comments
<ul>
    <li> 634 rows where there is missing data for players
    <li> The top 10 total strokes gained seasons was surprising with Rory McIlroy being the topy with $2.406$ in 2012
<ul>

## LaTeX
LateX is a typesetting programming language to display mathematical symbols.

### Examples
<ul>
    <li> $y = \beta_0 + \beta_1 x_1 + \beta_2 x_2$
    <li> $ \displaystyle \sum_{k=1}^{\infty} \frac{1}{k^2} = \frac{\pi^2}{6}$
<ul>